File: Dos4PeopleApp/wwwroot/Application_Scripts/ChangePassword.js


function MatchPassword() {
    if ($('#NewPassword').val() != $('#ConfirmPassword').val()) {
        toastr.warning("New Password and it's confirmation doesn't match")
    } 
}
function CheckExistingPassword() {
    if ($('#CurrentPassword').val() == '') {
        toastr.warning("Provide Password")
    }
    else {
        var objVmUser = {
            Password: $('#CurrentPassword').val()
        };
        $.ajax({
            url: "/ChangePassword/CheckExistingPassword",
            data: JSON.stringify(objVmUser),
            type: "POST",
            contentType: "application/json;charset=utf-8",
            success: function (response) {
                debugger;
                if (response.status == true) {

                } else {
                    $('#CurrentPassword').val('');
                    toastr.error(response.message);
                }
            },
            error: function (response) {
                toastr.error("error!");
            }
        });
    }
}


$('#btnChangePassword').click(function () {
   
    if ($('#CurrentPassword').val() == '') {
        toastr.warning("Provide Current Password")
    } 
    else if ($('#NewPassword').val() == '') {
        toastr.warning("Provide New Password")
    }
    else if ($('#ConfirmPassword').val() == '') {
        toastr.warning("Provide Confirm Password")
    } 
    else if ($('#NewPassword').val() != $('#ConfirmPassword').val()) {
        toastr.warning("New Password and it's confirmation doesn't match")
    } 
    else {
        var objVmUser = {
            CurrentPassword: $('#CurrentPassword').val(),
            Password: $('#NewPassword').val(),
            ConfirmPassword: $('#ConfirmPassword').val()           
        };
        $.ajax({
            url: "/ChangePassword/ChangeCurrentPassword",
            data: JSON.stringify(objVmUser),
            type: "POST",
            contentType: "application/json;charset=utf-8",
            success: function (response) {
                debugger;
                if (response.status == true) {
                    toastr.success(response.message);
                    Clear();
                } else {
                    toastr.error(response.message);
                }
            },
            error: function (response) {              
                toastr.error("error!"); 
            }
        });
    }
});

function Clear() {
    $('#CurrentPassword').val('');
    $('#NewPassword').val('');
    $('#ConfirmPassword').val('');   
}

File: Dos4PeopleApp/Controllers/ChangePassword.cs
using Dos4PeopleApp.DA;
using Dos4PeopleApp.Models;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Dos4PeopleApp.Utility;

namespace Dos4PeopleApp.Controllers
{
    public class ChangePassword : Controller
    {
        UserDA _objUserDa = null;
        VmUser ObjSession = null;
        public IActionResult Index()
        {           
            return View();
        }
        public ChangePassword()
        {
            _objUserDa = new UserDA();

        }
        public async Task<JsonResult> CheckExistingPassword([FromBody] VmUser objVmUser)
        {
            try
            {
                ObjSession = HttpContext.Session.GetObjectFromJson<VmUser>("VmUser");
                if (objVmUser.Password == null || objVmUser.Password == "")
                {
                    return Json(new { Message = "Provide Password", Status = false });
                }
                else
                {
                    VmUser _objUser = await _objUserDa.CheckExistingPassword(objVmUser.Password.Trim(), ObjSession.UserName);
                    return Json(new { Message = "", Status = true });
                }
            }
            catch (Exception ex)
            {
                return Json(new { Message = ex.Message, Status = false });
            }
        }
        public async Task<JsonResult> ChangeCurrentPassword([FromBody] VmUser objVmUser)
        {
            try
            {
                VmReturnType _objReturnType = null;
                ObjSession = HttpContext.Session.GetObjectFromJson<VmUser>("VmUser");               
                if (objVmUser.CurrentPassword == null || objVmUser.CurrentPassword == "")
                {
                    return Json(new { Message = "Provide Current Password", Status = false });
                }               
                else if (objVmUser.Password == null || objVmUser.Password == "")
                {
                    return Json(new { Message = "Provide New Password", Status = false });
                }
                else if (objVmUser.ConfirmPassword == null || objVmUser.ConfirmPassword == "")
                {
                    return Json(new { Message = "Provide Confirm Password", Status = false });
                }
                else if (objVmUser.Password != objVmUser.ConfirmPassword)
                {
                    return Json(new { Message = "New Password and it's confirmation doesn't match", Status = false });
                }
                else
                {
                    objVmUser.UserName = ObjSession.UserName;
                    _objReturnType = await _objUserDa.ChangeCurrentPassword(objVmUser);
                    return Json(new { Message = _objReturnType.UserMsg.Trim(), Status = _objReturnType.Status });
                }
            }
            catch (Exception ex)
            {
                return Json(new { Message = ex.Message, Status = false });
            }
        }
    }
}
